MovieApp/Views/HomeScreen/DefaultHomeScreen/HomeScreenTableViewCell.swift

```diff
@@ -25,40 +28,87 @@ class HomeScreenTableViewCell: UITableViewCell{
+                categoryTitle =  "For You"
+                fetchGenre()
+            default:
+                print("Wrong group")
         }
+    }
+    
+    private func fetchData(urlString: String) {
+        guard let url = URL(string: urlString) else {return}
+        
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
         
-        self.addSubview(moviePosters)
-        self.addSubview(categoryText)
+        let netowrkservice = NetworkService()
         
-        constraintViews()
+        netowrkservice.executeUrlRequest(request) { (result: Result<MovieList, RequestError>) in
+            switch result{
+            case .failure(let error):
+                print(error)
+            case .success(let value):
+                self.moviePosters = HorizontalMoviesView(movies: value)
+                self.addSubview(self.moviePosters)
+                self.moviePosters.selectMovieDelegate = self
+                
+                self.moviePosters.snp.makeConstraints(){
+                    $0.leading.trailing.equalToSuperview().inset(18)
+                    $0.top.equalToSuperview().offset(70)
+                    $0.height.equalTo(200)
+                }
+            }
+        }
     }
     
-    private func constraintViews(){
-        categoryText.snp.makeConstraints(){
-            $0.top.leading.trailing.equalToSuperview().inset(18)
-        }
+    private func fetchGenre() {
+        guard let url = URL(string: "https://api.themoviedb.org/3/genre/movie/list?language=en-US&api_key=225631dc133673048d4a801ff6951584") else {return}
+        
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
         
-        moviePosters.snp.makeConstraints(){
-            $0.leading.trailing.equalToSuperview().inset(18)
-            $0.top.equalTo(categoryText.snp.bottom).offset(5)
-            $0.height.equalTo(200)
+        let netowrkservice = NetworkService()
+        
+        netowrkservice.executeUrlRequest(request) { (result: Result<GenreList, RequestError>) in
+            switch result{
+            case .failure(let error):
+                print(error)
+            case .success(let value):
+                self.categoryText = MovieCategoryView(categoryTitleText: self.categoryTitle, genres: value)
+                self.addSubview(self.categoryText)
+                
+                self.categoryText.snp.makeConstraints(){
+                    $0.top.leading.trailing.equalToSuperview().inset(18)
+                    $0.height.equalTo(100)
+                }
+            }
         }
     }
+    
+    
+    func selectedMovieId(movieId: Int) {
+        selectedMovieDelegate2?.selectedMovieId(movieId: movieId)
+    }
+}
+
+protocol SelectedMovieDelegate2{
+    func selectedMovieId(movieId: Int)
 }
```

Comment:
There is no need to have multiple protocols here. You could just declare one protocol
"SelectedMovieDelegate" and use it everywhere.